bin-src/lib/comparePackageManifests.ts

```diff
@@ -0,0 +1,37 @@
+const compareObjectsShallowly = (objA = {}, objB = {}) => {
+  if (typeof objA !== typeof objB) {
+    return false;
+  }
+
+  if (typeof objA !== 'object' || typeof objB !== 'object' || objA === null || objB === null) {
+    return objA === objB;
+  }
+
+  const entriesA = Object.entries(objA).sort((a, b) => a[0].localeCompare(b[0]));
+  const entriesB = Object.entries(objB).sort((a, b) => a[0].localeCompare(b[0]));
+
+  if (entriesA.length !== entriesB.length) {
+    return false;
+  }
+
+  // depends on always having consistent ordering of keys
+  // eslint-disable-next-line no-plusplus
+  for (let i = 0; i < entriesA.length; i++) {
+    const [keyA, valueA] = entriesA[i];
+    const [keyB, valueB] = entriesB[i];
+
+    if (keyA !== keyB || valueA !== valueB) {
+      return false;
+    }
+  }
+
+  return true;
+};
+
+export const arePackageDependenciesEqual = (packageObjA, packageObjB) => {
+  const fields = ['dependencies', 'devDependencies', 'peerDependencies'];
```

Comment:
I know you mentioned about peerDependenciesMeta and resolutions, but I wanted to point possible important fields that are not here:
- optionalDependencies
- overrides (npm 8)
- resolutions (yarn)